File: 2. Add Two Numbers/main.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complementDict = {}
        for index, num in enumerate(nums):
            compliment = target - num
            if compliment in complementDict:
                return [complementDict[compliment], index]
            else:
                complementDict[num] = index
        return []
        

File: 2. Add Two Numbers/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut complement_map: HashMap<i32, i32> = HashMap::new();

        for (index, num) in nums.iter().enumerate() {
            let compliment = target - num;

            if let Some(i) = complement_map.get(&compliment) {
                return vec![i.to_owned(), index as i32];
            } else {
                complement_map.insert(num.clone(), index as i32);
            }
        };

        vec![]
    }
}

